docs/rfc/block-registration.md

```diff
@@ -0,0 +1,414 @@
+```
+
+This is a pointer to a php file returning a render callback php function.  The render callback is function called when the block is rendered on the frontend. It's used to generate the frontend markup dynamically.
+
+See the [dynamic blocks documentation](https://wordpress.org/gutenberg/handbook/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks/) for more details.
+
+### Styles Variations
+
+*   Type: `array`
+*   Optional
+*   Localized: No
+*   Property: `styleVariations`
+
+```json
+{ 
+	"styleVariations": [ 
+		{ "name": "default", "label": "Default", "isDefault": true }, 
+		{ "name": "other", "label": "Other" }
+	]
+}
+```
+
+Block styles can be used to provide alternative styles to block. It works by adding a class name to the block's wrapper. Using CSS, a theme developer can target the class name for the style variation if it is selected.
+
+Plugins and Themes can also register [custom block style](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#block-style-variations) for existing blocks.
+
+### Transforms
+
+* Type: `string`
+* Optional
+* Localized: No
+* Property: `transforms`
+
+```json
+{ "transforms": "my-block-transforms.js" }
+```
+
+This property is a pointer to a JavaScript file containing the save function of the block transforms. The save function defines the way in which the different attributes should be combined into the final markup, which is then serialized by Gutenberg into `post_content`.
+
+See the [Transforms](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#transforms-optional) documentation for more details.
+
+### Deprecated versions
+
+* Type: `object[]`
+* Optional
+* Localized: No
+* Property: `deprecated`
+
+```json
+{ "deprecated": [ {
+	"attributes": {},
+	"save": "my-deprecated-save.js",
+	"supports": {}
+} ] }
+```
+
+This property contains the definition of the deprecated versions of the block type. It is used to ensure that old blocks with old markup are not considered invalid.
+
+See the [Deprecated Blocks](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-deprecation/) documentation for more details.
+
+### Supports
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `supports`
+
+Optional block extended support features. 
+
+See the [block supports](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#supports-optional) documentation page for more details.
+
+### Stylesheets
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `stylesheets`
+
+```json
+{
+	"stylesheets": {
+		"main": "my-block-style.css",
+		"editor": "my-block-editor-style.css",
+		"theme": "my-block-theme-style.css"
+	}
+}
+```
+
+This property is a pointer to CSS files containing the CSS used for the block in different contexts.
+
+## Internationalization
```

Comment:
Updated according to the chat started in the weekly Core JS meeting last week (link requires registration):

https://wordpress.slack.com/archives/C5UNMSU4R/p1554216514081900